### DISCUSSION TITLE
using plot.momentuHMM in rmarkdown files

### DISCUSSION THREAD

**Discussion by justin-boldsen-ryan** [bmcclintock/momentuHMM#181] [Q&A] [answered]
Hi ya'll,

I'm having some issues with the plot.momentuHMM function while working within rmarkdown files. 

I'm not sure how to store the density/histogram plots from plot.momentuHMM as objects to use with cowplot. 

I'm looking to do something  like this: 

m1_warnd <- fitHMM(df_prepped, 
                   nbStates= n_states, 
                   dist= distributions, 
                   Par0=init_prob_distr,
                   stateNames = state_names)

p <- plot(m1_warnd, ask = FALSE) 

p1 <- p[[1]]
p2 <- p[[2]]

plot_grid(p1, p2, labels = c('A', 'B'), label_size = 12)

This code assumes that  plot.momentuHMM would returns a list containing all the plot objects. However, it seems to return a list with the bool value for the ask parameter. ultimately i'm looking to store the plot.momentuHMM plot objects as objects i can pass around in the code. Like in the above code.  Is there a way to get these objects to do this? 

Thanks!
Justin

**Comment by bmcclintock** [comment, marked as answer]
Before {momentuHMM} v2.0.0, you'd normally need to save the plots (e.g. as a PDF) and include them in your document using `\includegraphics`:

```
    ```{r example, echo=FALSE, results='hide'}
    pdf("example.pdf")
    plot(example$m, ask=FALSE)
    dev.off()
    ```
    \begin{figure}
        \centering
        \includegraphics[page=1,width=0.45\textwidth]{example.pdf}
        \includegraphics[page=2,width=0.45\textwidth]{example.pdf}
        \caption{Example}
        \label{fig:example}
    \end{figure}
```
However, with v2.0.0 (currently only available in the `develop` branch on [GitHub](https://github.com/bmcclintock/momentuHMM/tree/develop)), one can use the `return` argument to: 1) return the estimates, standard errors, and confidence intervals of the parameters as a function of any covariates (and create your own plots); and 2) return recorded plots (via `grDevices::recordPlot`) of the density/histogram and parameter plots.

Here's an example using the returned estimates:
```
remotes::install_github('bmcclintock/momentuHMM@develop')
library(momentuHMM)

out <- plot(example$m,plotCI=TRUE,plotStationary=TRUE,ask=FALSE,return=TRUE)

library(ggplot2)
library(patchwork)

p <- list()

for(i in names(out$estimates$beta$cov1)){
  p[[i]] <- ggplot(out$estimates$beta$cov1[[i]], aes(x=cov1, y=est))+
    geom_line() + geom_ribbon(aes(ymin=lci, ymax=uci),alpha = 0.15,colour=NA)+
    ylab(i) + ylim(0,1)
}

p$`1 -> 1` + p$`1 -> 2` + p$`2 -> 1` + p$`2 -> 2` + plot_layout(nrow=2,ncol=2,byrow=TRUE)

p2 <- list()

for(i in names(out$estimates$stationary$cov1)){
  p2[[i]] <- ggplot(out$estimates$stationary$cov1[[i]], aes(x=cov1, y=est))+
    geom_line() + geom_ribbon(aes(ymin=lci, ymax=uci),alpha = 0.15,colour=NA)+
    ylab(i) + ylim(0,1)
}

p2$`state 1` + p2$`state 2` + plot_layout(nrow=2,ncol=1,byrow=TRUE)
``` 
And here's an example using the recorded density/histogram plots:
```
library(cowplot)
plot_grid(out$plots$step$densityHist,out$plots$angle$densityHist, labels = c('A', 'B'), label_size = 12)
```
To create your own density/histogram plots, see #93.
